src/server/system_services/account_server.js

```diff
@@ -543,27 +543,24 @@ function update_account(req) {
  * RESET PASSWORD
  *
  */
-function reset_password(req) {
+async function reset_password(req) {
+    let is_authorized_account = await verify_authorized_account(req);
+    if (!is_authorized_account) {
+        throw new RpcError('UNAUTHORIZED', 'Invalid verification password');
+    }
     let account = system_store.data.accounts_by_email[req.rpc_params.email.unwrap()];
     if (!account) {
         throw new RpcError('NO_SUCH_ACCOUNT', 'No such account email: ' + req.rpc_params.email);
     }
     if (!is_support_or_admin_or_me(req.system, req.account, account)) {
```

Comment:
In order to detect if it's me, 
we can compare the provided username (email) to avoid getting the account details on this step,
and move the verification to the top